### ISSUE TITLE
plone.app.iterate - Content lost lock status after 10 minutes

### ISSUE THREAD

**Issue by agnogueira** [plone/Products.CMFPlone#1451]
When you make a checkout in any content, the user can not edit the original content until the user perform the checkin action in the copy. When accessing the original content, user will see a message telling that this content is locked and the edit action is unavailable.

But after 10 minutes, without any action, the content lost his locked status and the edit action becomes available.

![iterate](https://cloud.githubusercontent.com/assets/1030139/13646826/0842c9ea-e610-11e5-805d-71b6d4bb2ecc.png)
(same page, the second reloaded after 10 minutes)

I found this behavior in a clean Plone 4.3.6 install, only with plone.pp.iterate instaled.
How to reproduce:

1. Create a new Plone site;
2. Go to control panel an install plone.app.iterate;
3. Make a checkout for a page;
4. After 10 minutes, reload the original page.

**Comment by hvelarde** [comment]
@plone/framework-team can someone point us on where to look? we need to fix this for a couple of customers ASAP.

**Comment by davisagli** [comment]
Hmm, that's weird. 10 minutes is the default lock timeout in plone.locking (https://github.com/plone/plone.locking/blob/master/plone/locking/interfaces.py#L9), but plone.app.iterate is meant to use its own lock type which specifies the max timeout (https://github.com/plone/plone.app.iterate/blob/master/plone/app/iterate/interfaces.py#L45)

**Comment by rodfersou** [comment]
I think we are using wrong version of p.a.iterate:
```python
from zope.interface import Interface, Attribute
 27 from zope import schema
 28 
 29 from zope.component.interfaces import IObjectEvent
 30 from plone.locking.interfaces import LockType
 31 
 32 from Products.Archetypes.interfaces import IReference
 33 
 34 #################################
 35 ## Marker interface
 36 
 37 class IIterateAware( Interface ):
 38     """An object that can be used for check-in/check-out operations.
 39     """
 40 
 41 #################################
 42 ## Lock types
 43 
 44 ITERATE_LOCK = LockType( u'iterate.lock', stealable=False, user_unlockable=False )
```

**Comment by rodfersou** [comment]
```
$ ag iterate bin/instance
288:    '/srv/cache/eggs/plone.app.iterate-2.1.12-py2.7.egg',
```

**Comment by rodfersou** [comment]
```
$ ag locking bin/instance
41:    '/srv/cache/eggs/plone.locking-2.0.9-py2.7.egg',
```

**Comment by rodfersou** [comment]
@hvelarde let's try to update p.a.iterate to the last version :)

**Comment by rodfersou** [comment]
@davisagli is the 2.x version the recommended to Plone 4.x? if so the bug is still there:

https://github.com/plone/plone.app.iterate/blob/2.1.x/plone/app/iterate/interfaces.py#L44

**Comment by hvelarde** [comment]
@rodfersou in one of the customer we already have latest versions of any package possibly related with this issue:

```
$ git diff
diff --git a/buildout.d/versions.cfg b/buildout.d/versions.cfg
index d62bbb6..f5d551b 100644
--- a/buildout.d/versions.cfg
+++ b/buildout.d/versions.cfg
@@ -92,3 +92,8 @@ plone.keyring = 3.0.1
 plone.locking = 2.0.9
 plone.protect = 3.0.16
 plone4.csrffixes = 1.0.9
+
+plone.app.iterate = 2.1.14
+plone.app.stagingbehavior = 0.1
+plone.app.versioningbehavior = 1.1.4
+Products.CMFEditions = 2.2.18
```

**Comment by davisagli** [comment]
There's an entry in the plone.app.iterate changelog that says "setted lock timeout to MAX_TIMEOUT to avoid baseline unwanted unlock after 10 minutes [parruc]" in version 2.1.11 -- but it appears to have been added under the wrong heading, because github shows that the commit where it was added is only present in the 3.x series: https://github.com/plone/plone.app.iterate/commit/fdccb3695f23d1c89be6f8105f444662bc9b59f0

Timing out the lock after 10 minutes feels like a bug to me, so feel free to backport the fix to the 2.1.x branch.

**Comment by rodfersou** [comment]
Fixed in https://github.com/plone/plone.app.iterate/issues/23
